### PR TITLE
Use jsonb to avoid problems on batch delete

### PR DESCRIPTION
**Pull request by jordisala1991**
## Subject

I am targeting this branch, because this is the safe branch to do it.

## Changelog

```markdown
### Fixed
- Fixed usage of this bundle with PostgreSQL.
```

### CODE REVIEW

**Comment by jordisala1991** [comment]
@lukepass do you use MySQL? if so, can you check if this PR breaks your database schema in any way?

**Comment by jordisala1991** [comment]
Maybe it is better to just add the jsonb option to not do more hard BC breaks. @VincentLanglet 

(we have more usages of json fields in SonataPageBundle)
(The roles field will become an array again , that was a mistake when removing FOSUserBundle)

**Comment by lukepass** [comment]
Should I just go on this branch and launch migrations or make some media upload?

**Comment by jordisala1991** [comment]
Yes please, go to this branch and try to launch migrations. Also if you could remove your media table and try to update the schema again, to ensure nothing changes, would be helpful.

Thank you :)

File: src/Resources/config/doctrine/BaseMedia.orm.xml
Comment (ChristianGripp) [inline src/Resources/config/doctrine/BaseMedia.orm.xml:12]: Should we add tests for this?

**Review by core23 — APPROVED** [review]
Did not test this, but looks okay so far

**Comment by core23** [comment]
IMHO we can release the stable 4.0 release after merging this. The last RC release was in November.

**Comment by VincentLanglet** [comment]
Yes, I just wait for someone, like lukepass, to test this.

**Comment by lukepass** [comment]
Yes! Sorry monday I will test it. Thanks!

Il sab 22 gen 2022, 19:06 Vincent Langlet ***@***.***> ha
scritto:

File: src/Resources/config/doctrine/BaseMedia.orm.xml
Comment (jordisala1991) [reply src/Resources/config/doctrine/BaseMedia.orm.xml:12]: I don't think it is really easy to test. The problem with the distinct query is only when you do a batch delete.

I think we might start with testing on different databases instead of trying to reproduce the problem that this PR solves.

Anyway we shouldn't block the PR for its test IMO. I guess if it does not break with Sqlite we are fine.

Comment (VincentLanglet) [reply src/Resources/config/doctrine/BaseMedia.orm.xml:12]: I personally don't block the PR for unit/functional test, but for real tests.

I'd like to avoid to release the 4.0 with a major bug for mysql user and then be in the situation where reverting the change introduce a bug for mongo users.

**Comment by lukepass** [comment]
It generated a migration:

```
<?php

declare(strict_types=1);

namespace DoctrineMigrations;

use Doctrine\DBAL\Schema\Schema;
use Doctrine\Migrations\AbstractMigration;

/**
 * Auto-generated Migration: Please modify to your needs!
 */
final class Version20220124085858 extends AbstractMigration
{
    public function getDescription(): string
    {
        return '';
    }

    public function up(Schema $schema): void
    {
        // this up() migration is auto-generated, please modify it to your needs
        $this->addSql('ALTER TABLE media__media CHANGE provider_metadata provider_metadata JSON DEFAULT NULL');
    }

    public function down(Schema $schema): void
    {
        // this down() migration is auto-generated, please modify it to your needs
        $this->addSql('ALTER TABLE media__media CHANGE provider_metadata provider_metadata JSON DEFAULT NULL');
    }
}
```

I executed it and made some tests, it seems all ok! The schema didn't break and the previous data didn't corrupt.

**Review by VincentLanglet — APPROVED** [review]

**Pull request merged**
